### PR TITLE
Add broker features

### PR DESCRIPTION
**Pull request by rommeltorquator**
#### Add the very basic broker features

[WIP] Buyer basic funcionality

## The Story
I was able to successfully connect to the IEX API by creating a free account. I also build a basic layout for the navbar just to provide a basic navigation for our website. Here is what I did in the PR.
- Enhanced the navbar functionality.
- Added 3 different kinds of users that inherits the User model.
- Created a sign up and sign in page for a broker account.
- Created 2 controllers (Stocks and Home).
- Added routes for 2 controllers.
- Added views for 2 controllers.
- Added a user_id column in stocks model.
- Installed the IEX gem.
- Displayed the most active stocks.
- Enabled the broker to add stock on its portfolio.
- Enabled to delete a stock in a portfolio for broker.
- Displayed data for individual stock.
- Fixed the rubocop syntax errors.

## Screenshots
image
image
image
image

## Resources
- https://iexcloud.io/docs/api/#list
- https://qiita.com/alokrawat050/items/0d7791b3915579f95791
- https://github.com/dblock/iex-ruby-client#get-list

### CODE REVIEW

**Comment by jarencudilla** [comment]
Nice work, since we have our broker template setup now the real fun begins .

Approved!

**Review by jarencudilla — APPROVED** [review]
:)

File: app/controllers/stocks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  private
+
+  def fetch_api
+    @client = IEX::Api::Client.new(
+      # publishable_token: ENV['PUBLIC'],
+      # secret_token: ENV['SECRET'],
+      publishable_token: 'Tpk_5e637bd957564d788d24f9b9491c1576',
+      secret_token: 'Tsk_81e8f3dbee154676a9ece271e45ac5e3',
```
Comment (analyn-cajocson) [inline app/controllers/stocks_controller.rb]: remove here your credential

File: app/views/home/portfolio.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      <tr>
+        <th scope="row"><%= current_broker.id %></th>
+        <td><%= stock.name %></td>
+        <td><%= stock.symbol %></td>
+        <td><%= stock.current_price %></td>
+        <td><%= button_to "remove stock", stock_path(stock), method: :delete %></td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
```
Comment (analyn-cajocson) [inline app/views/home/portfolio.html.erb]: add new line at the end of file

File: app/views/partials/_header.html.erb [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+            <%= link_to "broker Sign up", new_broker_registration_path, class: "nav-link" %>
+          </li>
+          <li class="nav-item">
+            <%= button_to "broker Sign in", new_broker_session_path, class: "nav-link" %>
+          </li>
+        <% end %>  
+      </ul>
+    </div>
+  </div>
+</nav>
```
Comment (analyn-cajocson) [inline app/views/partials/_header.html.erb]: add new line at the end of file

File: app/views/stocks/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <h5 class="card-title"><%= stock.symbol %></h5>    
+        <h5 class="card-title"><%= stock.company_name %></h5>
+        <h5 class="card-title"><%= stock.iex_last_updated_t %></h5>
+        <p class="card-text"><%= stock.latest_price %></p>
+        <a href="#" class="btn btn-primary"><%= stock.avg_total_volume %></a>
+        <%= link_to "show stock", stock_path(stock.symbol), class: "btn btn-danger" %>
+      </div>
+    </div>
+  <% end %>
+</div>
```
Comment (analyn-cajocson) [inline app/views/stocks/index.html.erb]: add new line at the end of file

File: app/views/stocks/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+<h1>name: <%= @stock.company_name %></h1><br/>
+<h1>symbol: <%= @stock.symbol %></h1><br/>
+<h1>latest price: <%= number_to_currency(@stock.latest_price) %></h1><br/>
+<h1>change: <%= number_to_currency(@stock.change) %></h1><br/>
+<h1>change percent: <%= @stock.change_percent_s %></h1><br/>
+<h1>avg total volume: <%= @stock.avg_total_volume %></h1><br/>
+<h1>price as of: <%= @stock.latest_time %></h1><br/>
+
+<%= button_to "Add to portfolio", stocks_path(id: @stock.symbol), class: "btn btn-success", method: :post %>
```
Comment (analyn-cajocson) [inline app/views/stocks/show.html.erb]: add new line at the end of file

File: config/application.rb [hunk 6]
```diff
@@ -30,6 +37,6 @@ class Application < Rails::Application
     # the framework and any gems in your application.
 
     # Don't generate system test files.
-    config.generators.system_tests = nil
+    config.generators.system_tests = nil
```
Comment (analyn-cajocson) [inline config/application.rb:35]: remove the extra whitespace

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: app/views/home/portfolio.html.erb (hunk 2 above)
Comment (jarencudilla) [reply app/views/home/portfolio.html.erb]: rubocop -a removes those spaces in the EOL.
but rubocop also adds the space in the EOL itself it turn gives errors in the workflow.

that's why we always remove that new line after the EOL

File: config/application.rb (hunk 6 above)
Comment (analyn-cajocson) [reply config/application.rb:35]: @rommeltorquator

**Review by analyn-cajocson — APPROVED** [review]

**Pull request merged**